google-cloud-storage/src/test/java/com/google/cloud/storage/it/StorageNativeCanary.java

```diff
@@ -0,0 +1,187 @@
+import org.junit.Test;
+
+// Intentionally avoid StorageITRunner here. It touches lots of code at a semi-static level making
+// native-test have a hard time.
+public final class StorageNativeCanary {
+
+  private static final int _512KiB = 512 * 1024;
+  private static final int _256KiB = 256 * 1024;
+  private static final byte[] bytes = DataGenerator.base64Characters().genBytes(_512KiB);
+
+  @Test
+  public void canary_happyPath_http() throws Exception {
+    canary_happyPath(StorageOptions.http().build().getService());
+  }
+
+  @Test
+  public void canary_happyPath_grpc() throws Exception {
+    canary_happyPath(StorageOptions.grpc().build().getService());
+  }
+
+  /**
+   * When testing on Native Image, we're primarily wanting to verify the primary code paths are
+   * properly detected by the native image compiler.
+   *
+   * <p>For Storage, we have a few "primary code paths" we want to ensure are validated:
+   *
+   * <ul>
+   *   <li>Can a (Unary) Request Succeed?
+   *   <li>Can a (ServerStream) Object Read Request Succeed?
+   *   <li>Can a (ClientStream) Object Write Request Succeed?
+   *   <li>Can a (Page over Unary) Paginated Request Succeed?
+   * </ul>
+   *
+   * To validate this, our happy path test is as follows:
+   *
+   * <ul>
+   *   <li>Create a temporary bucket (Unary)
+   *   <li>Insert two (2) objects (Unary, ServerStream)
+   *   <li>List all objects, using a pageSize of 1 (Page over Unary)
+   *   <li>Read all bytes of each object (ServerStream)
+   *   <li>Delete each object (Unary)
+   *   <li>Delete temporary bucket (Unary)
+   * </ul>
+   */
+  private static void canary_happyPath(Storage storage) throws Exception {
+    // create a temporary bucket
+    try (TemporaryBucket tempB =
+        TemporaryBucket.newBuilder()
+            .setStorage(storage)
+            .setBucketInfo(BucketInfo.of("java-storage-grpc-" + UUID.randomUUID()))
+            .build()) {
+      String bucketName = tempB.getBucket().getName();
+      String obj1Name = UUID.randomUUID().toString();
+      String obj2Name = UUID.randomUUID().toString();
+
+      // insert 2 objects
+      BlobInfo info1 = BlobInfo.newBuilder(bucketName, obj1Name).build();
+      BlobInfo info2 = BlobInfo.newBuilder(bucketName, obj2Name).build();
+      uploadUsingWriter(storage, info1);
+      uploadUsingWriter(storage, info2);
+
+      // list objects
+      Page<Blob> page = storage.list(bucketName, BlobListOption.pageSize(1));
+      List<Blob> blobs = ImmutableList.copyOf(page.iterateAll());
+
+      // read all bytes of each object
+      List<BlobWithContent> actual =
+          blobs.stream()
+              .map(info -> readAll(storage, info))
+              .collect(ImmutableList.toImmutableList());
+
+      List<Boolean> deletes =
+          blobs.stream()
+              .map(b -> storage.delete(b.getBlobId(), BlobSourceOption.generationMatch()))
+              .collect(ImmutableList.toImmutableList());
+
+      assertAll(
```

Comment:
This is a util method we've added in storage that ensures all of the assertions contained within it are ran and collected if they fail. This will report multiple failures if multiple assertions fail.

Reply:
Thank you!